File: keyence2019b.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <utility>
#include <string>
#include <queue>
#include <stack>

using namespace std;

typedef long long int ll;
typedef pair<int, int> Pii;

const ll mod = 1000000007;

int main() {
  cin.tie(0);
  ios::sync_with_stdio(false);

  string s;
  cin >> s;

  int n = s.length();

  string keyence = "keyence";
  int p = 0;
  for (int i = 0; i < 7; i++) {
    if (s[i] == keyence[p]) p++;
    else break;
  }
  for (int i = n-7+p; i < n; i++) {
    if (s[i] == keyence[p]) p++;
    else break;
  }

  if (p == 7) cout << "YES" << endl;
  else cout << "NO" << endl;

  return 0;
}

File: keyence2019b.cs
using System;

public class keyence2019b {
  static public void Main(){
    string s = Console.ReadLine();

    int n = s.Length;

    string keyence = "keyence";
    for (int i = 0; i <= 7; i++) {
      if (s.Substring(0, i) == keyence.Substring(0, i) && s.Substring(n-7+i, 7-i) == keyence.Substring(i, 7-i)) {
        Console.WriteLine("YES");
        return;
      }
    }

    Console.WriteLine("NO");

    return;
  }
}
